src/topology/sheaves/structure_sheaf.lean

```diff
@@ -0,0 +1,214 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin, Scott Morrison
+-/
+import algebraic_geometry.prime_spectrum
+import ring_theory.localization
+import algebra.category.CommRing
+import topology.sheaves.local_predicate
+import topology.sheaves.forget
+import ring_theory.bundled_subring
+
+/-!
+# The structure sheaf on `prime_spectrum R`.
+
+-/
+
+universe u
+
+noncomputable theory
+
+variables (R : Type u) [comm_ring R]
+
+open Top
+open topological_space
+open category_theory
+open opposite
+
+/--
+The type family over `prime_spectrum R` consisting of the localization over each point.
+-/
+@[reducible]
```

Comment:
Does this need to be reducible, or can you `[derive comm_ring, derive local_ring]`?

Reply:
The problem is this is the type family itself, so it doesn't make sense to put `comm_ring` structure on it.

Reply:
Oooh, but we can move `P` left of the colon.

Reply:
No --- it's a lambda, not a pi.

Reply:
I decided to use derive handlers anyway.

Reply:
You're absolutely right -- I was mistaken!